manifests/trusted_external_command.pp

```diff
@@ -69,7 +69,10 @@
     },
   ])
 
-  unless Service['pe-puppetserver'] {
+  unless defined(Service['pe-puppetserver']) {
```

Comment:
@reidmv Do you think we could remove the `defined` check and just declare the `puppetserver` resource? I remember you mentioning that doing that could lead to a slight performance hit, so wanted to double check.

Reply:
This stanza should be fully removed, and the `pe-puppetserver` resource should **not** be declared.

The `defined()` function can reliably determine if a class definition is available (whether or not it exists and could be included), but `defined()` is unreliable for determining if a resource _will be_ present in the catalog—it is evaluation-order dependent. If the `puppet_enterprise::master` class has been evaluated first, then this `defined()` would return `true`. If the `puppet_enterprise::master` class has not been evaluated _yet_, this `defined()` will return false, the resource will be declared, and when `puppet_enterprise::master` is evaluated later, a duplicate-resource-definition error will be raised.

This class should depend on and require that something else declare `Service[pe-puppetserver]`. It should never declare that service itself.

There are other ways this can be addressed too but IMO the extra complexity and parameterization (or resource collector tricks) do not sufficiently improve the UX to be worth the complexity cost.

Reply:
Thanks for the feedback. Will remove.